ci/steps/publish-npm-dev.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
```

Comment:
❓Should we refactor [`publish-npm.sh` script](https://github.com/coder/code-server/blob/main/ci/steps/publish-npm.sh) instead of adding this script? They're quite similar.

Reply:
That does seem worthwhile to me.  Could tackle that later though.

Reply:
Eh I'll do it in this one! Shouldn't be too difficult